src/CDSAPI.jl

```diff
@@ -1,88 +1,116 @@
+        fname::AbstractString,
+        keys::AbstractDict = cdskeys()
+    )
+
+Retrieves datasets from the Climate Data Store, with options specified in a Julia Dictionary and saves it into a specified file.
+
+Arguments:
+    * `dataset::AbstractString` : string specifies the name of the dataset within the Climate Data Store that the `retrieve` function is attempting to retrieve data from
+    * `params::AbstractDict` : dictionary that contains the keywords that specify the properties (e.g. date, resolution, grid) of the data being retrieved
+    * `fname::AbstractString` : string that contains the path and name of the file that the data is to be saved into
+    * `keys::AbstractDict` : dictionary that contains API Key information read from the .cdsapirc file in the home directory (optional)
 """
-function retrieve(name, params, filename)
-    creds = Dict()
-    open(joinpath(homedir(),".cdsapirc")) do f
-        for line in readlines(f)
-            key, val = strip.(split(line,':', limit=2))
-            creds[key] = val
-        end
-    end
+function retrieve(
+    dataset::AbstractString,
+    params::AbstractDict,
+    fname::AbstractString,
+    keys::AbstractDict = cdskeys()
+)
 
-    apikey = string("Basic ", base64encode(creds["key"]))
+    @info "$(now()) - Welcome to the Climate Data Store"
+    apikey = string("Basic ", base64encode(keys["key"]))
+
+    @info "$(now()) - Sending request to https://cds.climate.copernicus.eu/api/v2/resources/$(dataset) ..."
     response = HTTP.request(
-        "POST",
-        creds["url"] * "/resources/$name",
+        "POST", keys["url"] * "/resources/$(dataset)",
         ["Authorization" => apikey],
         body=JSON.json(params),
-        verbose=1)
-
+        verbose=0
+    )
     resp_dict = JSON.parse(String(response.body))
     data = Dict("state" => "queued")
-    while data["state"] != "completed"
-        data = HTTP.request("GET", creds["url"] * "/tasks/" * string(resp_dict["request_id"]),  ["Authorization" => apikey])
+
+    @info "$(now()) - Request is queued"
+    while data["state"] == "queued"
+        data = HTTP.request(
+            "GET", keys["url"] * "/tasks/" * string(resp_dict["request_id"]),
+            ["Authorization" => apikey]
+        )
         data = JSON.parse(String(data.body))
-        println("request queue status ", data["state"])
     end
 
-    HTTP.download(data["location"], filename)
-    return data
+    @info "$(now()) - Request is running"
+    while data["state"] == "running"
+        data = HTTP.request(
+            "GET", keys["url"] * "/tasks/" * string(resp_dict["request_id"]),
+            ["Authorization" => apikey]
+        )
+        data = JSON.parse(String(data.body))
+    end
+
+    if data["state"] == "completed"
+
+        @info "$(now()) - Request is completed"
+
+        @info """$(now()) - Downloading $(uppercase(dataset)) data
+          $(BOLD("URL:"))         $(data["location"])
+          $(BOLD("Destination:")) $(fnc)
```

Comment:
Also @natgeo-wong, while calling `retrieve()`, it is throwing `UndefVarError: BOLD not defined` error. Am I missing something?

Reply:
Damn, forgot about Crayons.Box HAHA.  gimme a minute, will add the verbose option and the Crayons package.